Sming/SmingCore/Network/TcpConnection.cpp

```diff
@@ -82,8 +82,11 @@ bool TcpConnection::connect(String server, int port, boolean useSsl /* = false *
 
 bool TcpConnection::connect(IPAddress addr, uint16_t port, boolean useSsl /* = false */, uint32_t sslOptions /* = 0 */)
 {
+	if (tcp == NULL)
+		initialize(tcp_new());
+
+  this->useSsl = useSsl;
```

Comment:
If I am not mistaken this line has to be inside of the #ifdef ENABLE_SSL block. Not before it.